src/api/client.ts

```diff
@@ -49,11 +50,21 @@ function shouldRetry(error: AxiosError): boolean {
   const shouldForceRefresh = error.config?.forceRefreshIdToken;
   if (shouldForceRefresh) return true;
 
-  const shouldRetryOnNetworkErrorOrIdempotentRequest =
+  return (
     Boolean(error.config?.retry) &&
     (getAxiosErrorType(error) === 'network-error' ||
-      isIdempotentRequestError(error));
-  return shouldRetryOnNetworkErrorOrIdempotentRequest;
+      isIdempotentRequestError(error))
+  );
+}
+
+// Implements a custom axios-retry to force a refresh if 401 error is found
+function handleUnauthorizedError(error: AxiosError) {
+  if (error.config && error.response?.status === 401) {
+    error.config.forceRefreshIdToken = true;
+    SHOULD_FORCE_REFRESH.value = true;
+    return client(error.config);
```

Comment:
Not really sure about returning a client invocation here 🤔 Could you sum up some reasoning behind it?